taca/illumina/__init__.py

```diff
@@ -0,0 +1,172 @@
+        if cl_options.get('interop-dir'):
+            cl.extend(['--interop-dir', cl_options.get('interop-dir')])
+        if cl_options.get('stats-dir'):
+            cl.extend(['--stats-dir', cl_options.get('stats-dir')])
+        if cl_options.get('reports-dir'):
+            cl.extend(['--reports-dir', cl_options.get('reports-dir')])
+
+        # Processing cl_options
+        threads = cl_options.get('loading-threads')
+        if threads and type(threads) is int:
+            cl.extend(['--loading-threads', '{}'.format(threads)])
+        threads = cl_options.get('demultiplexing-threads')
+        if threads and type(threads) is int:
+            cl.extend(['--demultiplexing-threads', '{}'.format(threads)])
+        threads = cl_options.get('processing-threads')
+        if threads and type(threads) is int:
+            cl.extend(['--processing-threads', '{}'.format(threads)])
+        threads = cl_options.get('writing-threads')
+        if threads and type(threads) is int:
+            cl.extend(['--writing-threads', '{}'.format(threads)])
+
+        # Behavioral options
+        adapter_stringency = cl_options.get('adapter-stringency')
+        if adapter_stringency and type(adapter_stringency) is float:
+            cl.extend(['--adapter-stringency', adapter_stringency])
+        aggregated_tiles = cl_options.get('aggregated-tiles')
+        if aggregated_tiles and aggregated_tiles in ['AUTO', 'YES', 'NO']:
+            cl.etend(['--aggregated-tiles', aggregated_tiles])
+        barcode_missmatches = cl_options.get('barcode-missmatches')
+        if barcode_missmatches and type(barcode_missmatches) is int \
+                and barcode_missmatches in range(3):
+            cl.extend(['--barcode-missmatches', barcode_missmatches])
+        if cl_options.get('create-fastq-for-index-reads'):
+            cl.append('--create-fastq-for-index-reads')
+        if cl_options.get('ignore-missing-bcls'):
+            cl.append('--ignore-missing-bcls')
+        if cl_options.get('ignore-missing-filter'):
+            cl.append('--ignore-missing-filter')
+        if cl_options.get('ignore-missing-locs'):
+            cl.append('--ignore-missing-locs')
+        mask = cl_options.get('mask-short-adapter-reads')
+        if mask and type(mask) is int:
+            cl.extend(['--mask-short-adapter-reads', mask])
+        minimum = cl_options.get('minimum-trimmed-reads')
+        if minimum and type(minimum) is int:
+            cl.extend(['--minimum-trimmed-reads', minimum])
+        if cl_options.get('tiles'):
+            cl.extend(['--tiles', cl_options.get('tiles')])
+
+        if cl_options.get('use-bases-mask'):
+            cl.extend(['--use-bases-mask', cl_options.get('use-bases-mask')])
+
+        if cl_options.get('with-failed-reads'):
+            cl.append('--with-failed-reads')
+        if cl_options.get('write-fastq-reverse-complement'):
+            cl.append('--write-fastq-reverse-complement')
```

Comment:
Hej @guillermo-carrasco a suggestion/question, instead of adding these `if` conditions to check for `bcl2fastq` and update the `command`. Cant we directly add the exact commands in the `config file` ? I mean additional parameter is only going to be added if its mentioned in `config` file. Hence we already reply on the config file, so why can't we take them(parameters) directly from it(config). So for example if the config looks something like this

```
bcl2fastq:
            path: /path/to/bcl2fastq
            XTen: /path/to/bcl2fastq (XTen version)
            HiSeq: /path/to/bcl2fastq (HiSeq version)
            MiSeq: /path/to/bcl2fastq (MiSeq version)
            parameters:
                    --runfolder path/to/run
                    --output-dir Demultiplexing
                    --loading-threads 2
                    --with-failed-reads
                    --stats-dir /some/path
```
and can we something do like this maybe 
```
with chdir(run):
        cl_options = config['bcl2fastq']
        cl = [cl_options.get('XTen')]

        #add given parameters from config
        bcl_params = cl_options.get('parameters','').split()
        if bcl_params:
            cl.extend(bcl_params)
        else:
            LOG.warn("Thats strange no bcl2fastq parameters where given in the config file!!")
            return
```

Maybe you have good reason for not to do so, but I am just curious :)